feat(curriculum): add maps and sets workshop to FSD cert (#57189)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/workshop-plant-nursery-catalog/6738ad2fd11775752d23ddc2.md

---
id: 6738ad2fd11775752d23ddc2
title: Step 32
challengeType: 0
dashedName: step-32
---

# --description--

The `Set` constructor also accepts an iterable as its argument and creates a set from it:

```js
const myArr = [ 4, 8, 4, 15, 16, 23, 23, 42, 15];
const mySet = new Set(myArr); // Set(6) [ 4, 8, 15, 16, 23, 42 ]
```

> To complete this workshop, modify your `displayPlantsSet` to return a set with the unique plant common names in your catalog.

Reviewer: ```suggestion
To complete this workshop, modify your `displayPlantsSet` to return a set of the unique plant common names in your catalog.
```

> To achieve it, create an array containing the `commonName` of each plant object stored as a key of the `catalog` map. Pass the array to the `Set` constructor and return the set from the function.

Reviewer: ```suggestion
To achieve this, create an array of unique `commonName` values from your `catalog` keys. Pass the array to the `Set` constructor and return the set from the function.
```

I'm not sure if this is any better. Here's the solution I came up with:

<details><summary>code</summary>

```js
const displayPlantsSet = () => {
    const uniqueCommonNames = [];

    catalog.forEach((val, key) => {
        if (!uniqueCommonNames.includes(key.commonName)) {
            uniqueCommonNames.push(key.commonName)
        }
    })

    const catalogSet = new Set(uniqueCommonNames);
    return catalogSet
};
```

</details>

Could maybe mention to remove the `add` methods.

Author: The intention was to create an array with all the common names, without caring about repetitions. Then use the set constructor to get the unique values. Anyway I'm checking just the final result.
